android/sandwich/src/main/kotlin/io/qonversion/sandwich/QonversionSandwich.kt

```diff
@@ -143,10 +110,12 @@ class QonversionSandwich(
             override fun onSuccess(products: Map<String, QProduct>) {
                 val product = products[productId]
                 val oldProduct = products[oldProductId]
-                if (product != null && oldProduct != null) {
-                    val purchaseOptions = configurePurchaseOptions(offerId, applyOffer, oldProduct, updatePolicyKey, contextKeys)
+                val purchaseOptions = configurePurchaseOptions(offerId, applyOffer, oldProduct, updatePolicyKey, contextKeys)
 
+                if (product != null && oldProduct != null) {
                     Qonversion.shared.updatePurchase(currentActivity, product, purchaseOptions, purchaseCallback)
+                } else if (product != null) {
+                    Qonversion.shared.purchase(currentActivity, product, purchaseOptions, purchaseCallback)
                 } else {
                     purchaseCallback.onError(QonversionError(QonversionErrorCode.ProductNotFound))
                 }
```

Comment:
```suggestion
                if (product == null) {
                    purchaseCallback.onError(QonversionError(QonversionErrorCode.ProductNotFound))
                    return
                }

                if (oldProduct == null) {
                    Qonversion.shared.purchase(currentActivity, product, purchaseOptions, purchaseCallback)
                } else {
                    Qonversion.shared.updatePurchase(currentActivity, product, purchaseOptions, purchaseCallback)
                }
```

i think that's more clear way

Reply:
We decided to leave it as it is.